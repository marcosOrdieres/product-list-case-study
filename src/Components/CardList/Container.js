import styled from 'styled-components';

const Container = styled.div`
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  box-shadow: rgba(12, 15, 20, 0.02) 0px 0px 0px 1px,
    rgba(12, 15, 20, 0.06) 0px 0px 1px 0px,
    rgba(12, 15, 20, 0.06) 0px 2px 2px 0px;
  border-radius: 10px;
  padding: 0px;
  margin-bottom: 10px;
`;

Container.displayName = 'CardList.Container';

export default Container;