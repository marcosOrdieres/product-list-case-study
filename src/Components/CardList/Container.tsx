import styled from 'styled-components';

const Container = styled.div`
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  box-shadow: rgba(12, 15, 20, 0.2) 0px 0px 0px 1px,
    rgba(12, 15, 20, 0.6) 0px 0px 1px 0px,
    rgba(12, 15, 20, 0.6) 0px 2px 2px 0px;
  border-radius: 10px;
  padding: 0px;
  margin-bottom: 30px;
`;

Container.displayName = 'CardList.Container';

export default Container;