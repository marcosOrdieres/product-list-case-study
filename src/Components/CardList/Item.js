import styled from 'styled-components';

const Item = styled.div`
  align-items: center;
  position: relative;
  border-bottom: 1px solid 5px;
  padding: 5px;

  &:first-child {
    border-top-left-radius: 1px;
    border-top-right-radius: 1px;
  }

  &:last-child {
    border-bottom-left-radius: 1px;
    border-bottom-right-radius: 1px;
  }
`;

Item.displayName = 'CardList.Item';

export default Item;
