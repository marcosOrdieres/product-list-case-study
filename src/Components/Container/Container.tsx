import styled from 'styled-components';

export interface ContainerProps {
  children?: any;
}

const ContainerStyled = styled.div`
  display: flex;
  flex-direction:column;
  width: 100vw;
  height: 100vh;
  align-items: center;
`;

export const Conatainer: React.FC<ContainerProps> = (props) => <ContainerStyled {...props}>{props.children}</ContainerStyled>;

export default ContainerStyled;