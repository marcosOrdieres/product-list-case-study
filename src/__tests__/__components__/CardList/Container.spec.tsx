import { render } from '@testing-library/react';
import React from 'react';
import Container from '../../../Components/CardList/Container';
import Item from '../../../Components/CardList/Item';


describe('Card List Container', () => {
  it('should render matching default styles', () => {
    const { asFragment } = render(
      <Container>
        <Item>1</Item>
        <Item>2</Item>
        <Item>3</Item>
      </Container>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
