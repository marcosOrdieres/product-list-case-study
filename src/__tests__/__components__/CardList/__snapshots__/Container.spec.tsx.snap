// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card List Container should render matching default styles 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBQB lezuDS"
  >
    <div
      class="sc-gsTEea MCeuV"
    >
      1
    </div>
    <div
      class="sc-gsTEea MCeuV"
    >
      2
    </div>
    <div
      class="sc-gsTEea MCeuV"
    >
      3
    </div>
  </div>
</DocumentFragment>
`;
